Octopus.Client/PowerShell/Variables/AddLibraryVariableSetToSelectedProjects.ps1

```diff
@@ -0,0 +1,45 @@
+# You can this dll from your Octopus Server/Tentacle installation directory or from
+# https://www.nuget.org/packages/Octopus.Client/
+Add-Type -Path 'Octopus.Client.dll' 
+ 
+$octopusProjectName = @() # Name of the project(s) you want to add/remove the variable set
+$octopusVariableSetName = "" # !! CASE SENSATIVE !! Which variable set do you want to add/remove?
+$apikey = 'API-ABCDEFGHIJKLMNOPQ' # Get this from your profile
+$octopusURI = 'https://yourplace.octopus.app' # Your server address
+
+$endpoint = New-Object Octopus.Client.OctopusServerEndpoint $octopusURI,$apikey 
+$repository = New-Object Octopus.Client.OctopusRepository $endpoint
+
+# Gets all existing variable sets from Octopus
+$allVariableSets = $repository.LibraryVariableSets.FindAll()
+
+# Gets the index of the variable set that we want to add
+$indexOfVariableSetIAmUsing = $allVariableSets.name.IndexOf($octopusVariableSetName)
+
+# Gets the object for the one variable set that we want
+$libraryVariableSet = New-Object Octopus.Client.Model.LibraryVariableSetResource
+$libraryVariableSet = $repository.LibraryVariableSets.Get($allVariableSets[$indexOfVariableSetIAmUsing].Id)
+
+# I don't really know what this does
+$ignoreConfigTransformVariable = new-object Octopus.Client.Model.VariableResource
+$ignoreConfigTransformVariable.Name = "Octopus.Action.Package." + $libraryVariableSet.Name
+$ignoreConfigTransformVariable.Value = "true"
+
+$variables = $repository.VariableSets.Get($libraryVariableSet.VariableSetId)
+$variables.Variables.Add($ignoreConfigTransformVariable)
+$repository.VariableSets.Modify($variables)
+
+$projects = @()
+# Find the project(s) you want
+foreach($name in $octopusProjectName){
+    $allProjects = $repository.Projects.FindAll()
```

Comment:
Why not do this in a single loop and use `$repository.Projects.FindByName($name)` rather than iterating names and then projects? `FindAll` is paged, so while it may seem more efficient to do this in one call for someone with a lot of projects it is very inefficient.

Reply:
Because I'm an idiot of course! I used FindByName everywhere else, but I guess I left this one alone.